### ISSUE TITLE
I can't verify the result

### ISSUE THREAD

**Issue by ZHIBINJIAN** [iMoonLab/yolov13#20]
I followed the official tutorial. 1. Train coco128. I started training map=0 from 0. 2. If I load the pre-trained weights of yolov13, the map would be 0.7, but I don't know if the network is trained. 3. Then I used my own dataset to load the yolov13 pre-trained weights, with a map of 0.000 or so, but it couldn't be improved. Why is this?

PS C:\Users\29363\Desktop\yolov13-main> & C:/Users/29363/AppData/Local/conda/conda/envs/yolov13/python.exe c:/Users/29363/Desktop/yolov13-main/train.py
New https://pypi.org/project/ultralytics/8.3.159 available 😃 Update with 'pip install -U ultralytics'
Ultralytics 8.3.63 🚀 Python-3.11.13 torch-2.7.1+cu128 CUDA:0 (NVIDIA GeForce RTX 5070 Laptop GPU, 8151MiB)
engine\trainer: task=detect, mode=train, model=yolov13s.yaml, data=coco128.yaml, epochs=20, time=None, patience=100, batch=8, imgsz=640, save=True, save_period=-1, cache=False, device=None, workers=4, project=None, name=train16, exist_ok=False, pretrained=True, optimizer=auto, verbose=True, seed=0, deterministic=True, single_cls=False, rect=False, cos_lr=False, close_mosaic=10, resume=False, amp=True, fraction=1.0, profile=False, freeze=None, multi_scale=False, overlap_mask=True, mask_ratio=4, dropout=0.0, val=True, split=val, save_json=False, save_hybrid=False, conf=None, iou=0.7, max_det=300, half=False, dnn=False, plots=True, source=None, vid_stride=1, stream_buffer=False, visualize=False, augment=False, agnostic_nms=False, classes=None, retina_masks=False, embed=None, show=False, save_frames=False, save_txt=False, save_conf=False, save_crop=False, show_labels=True, show_conf=True, show_boxes=True, line_width=None, format=torchscript, keras=False, optimize=False, int8=False, dynamic=False, simplify=True, opset=None, workspace=None, nms=False, lr0=0.01, lrf=0.01, momentum=0.937, weight_decay=0.0005, warmup_epochs=3.0, warmup_momentum=0.8, warmup_bias_lr=0.0, box=7.5, cls=0.5, dfl=1.5, pose=12.0, kobj=1.0, nbs=64, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, degrees=0.0, translate=0.1, scale=0.5, shear=0.0, perspective=0.0, flipud=0.0, fliplr=0.5, bgr=0.0, mosaic=1.0, mixup=0.0, copy_paste=0.1, copy_paste_mode=flip, auto_augment=randaugment, erasing=0.4, crop_fraction=1.0, cfg=None, tracker=botsort.yaml, save_dir=runs\detect\train16

                   from  n    params  module                                       arguments
  0                  -1  1       928  ultralytics.nn.modules.conv.Conv             [3, 32, 3, 2]
  1                  -1  1      9344  ultralytics.nn.modules.conv.Conv             [32, 64, 3, 2, 1, 2]
  2                  -1  1     20800  ultralytics.nn.modules.block.DSC3k2          [64, 128, 1, False, 0.25]
  3                  -1  1     37120  ultralytics.nn.modules.conv.Conv             [128, 128, 3, 2, 1, 4]
  4                  -1  1     78464  ultralytics.nn.modules.block.DSC3k2          [128, 256, 1, False, 0.25]
  5                  -1  1     68352  ultralytics.nn.modules.conv.DSConv           [256, 256, 3, 2]
  6                  -1  2    677120  ultralytics.nn.modules.block.A2C2f           [256, 256, 2, True, 4]        
  7                  -1  1    134400  ultralytics.nn.modules.conv.DSConv           [256, 512, 3, 2]
  8                  -1  2   2664960  ultralytics.nn.modules.block.A2C2f           [512, 512, 2, True, 1]        
  9           [4, 6, 8]  1   1343744  ultralytics.nn.modules.block.HyperACE        [256, 256, 1, 8, True, True, 0.5, 1, 'both']
 10                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']
 11                   9  1    132096  ultralytics.nn.modules.block.DownsampleConv  [256]
 12              [6, 9]  1         1  ultralytics.nn.modules.block.FullPAD_Tunnel  []
 13             [4, 10]  1         1  ultralytics.nn.modules.block.FullPAD_Tunnel  []
 14             [8, 11]  1         1  ultralytics.nn.modules.block.FullPAD_Tunnel  []
 15                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']
 16            [-1, 12]  1         0  ultralytics.nn.modules.conv.Concat           [1]
 17                  -1  1    443648  ultralytics.nn.modules.block.DSC3k2          [768, 256, 1, True]
 18             [-1, 9]  1         1  ultralytics.nn.modules.block.FullPAD_Tunnel  []
 19                  17  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']
 20            [-1, 13]  1         0  ultralytics.nn.modules.conv.Concat           [1]
 21                  -1  1    131712  ultralytics.nn.modules.block.DSC3k2          [512, 128, 1, True]
 22                  10  1     33024  ultralytics.nn.modules.conv.Conv             [256, 128, 1, 1]
 23            [21, 22]  1         1  ultralytics.nn.modules.block.FullPAD_Tunnel  []
 24                  -1  1    147712  ultralytics.nn.modules.conv.Conv             [128, 128, 3, 2]
 25            [-1, 18]  1         0  ultralytics.nn.modules.conv.Concat           [1]
 26                  -1  1    345344  ultralytics.nn.modules.block.DSC3k2          [384, 256, 1, True]
 27             [-1, 9]  1         1  ultralytics.nn.modules.block.FullPAD_Tunnel  []
 28                  26  1    590336  ultralytics.nn.modules.conv.Conv             [256, 256, 3, 2]
 29            [-1, 14]  1         0  ultralytics.nn.modules.conv.Concat           [1]
 30                  -1  1   1346048  ultralytics.nn.modules.block.DSC3k2          [768, 512, 1, True]
 31            [-1, 11]  1         1  ultralytics.nn.modules.block.FullPAD_Tunnel  []
 32        [23, 27, 31]  1    850368  ultralytics.nn.modules.head.Detect           [80, [128, 256, 512]]
YOLOv13s summary: 648 layers, 9,055,527 parameters, 9,055,511 gradients, 21.2 GFLOPs

Freezing layer 'model.32.dfl.conv.weight'
AMP: running Automatic Mixed Precision (AMP) checks...
AMP: checks passed ✅
train: Scanning C:\Users\29363\Desktop\yolov13-main\datasets\coco128\labels\train2017.cache... 126 images, 2 backgrounds, 0 corrupt: 100%|██████████| 128/128 [00:00<?, ?it/s]
albumentations: Blur(p=0.01, blur_limit=(3, 7)), MedianBlur(p=0.01, blur_limit=(3, 7)), ToGray(p=0.01, method='weighted_average', num_output_channels=3), CLAHE(p=0.01, clip_limit=(1.0, 4.0), tile_grid_size=(8, 8)), ImageCompression(p=0.5, compression_type='jpeg', quality_range=(75, 100))
val: Scanning C:\Users\29363\Desktop\yolov13-main\datasets\coco128\labels\train2017.cache... 126 images, 2 backgrounds, 0 corrupt: 100%|██████████| 128/128 [00:00<?, ?it/s]
C:\Users\29363\AppData\Local\conda\conda\envs\yolov13\Lib\site-packages\albumentations\check_version.py:147: UserWarning: Error fetching version info The read operation timed out
  data = fetch_version_info()
Plotting labels to runs\detect\train16\labels.jpg... 
optimizer: 'optimizer=auto' found, ignoring 'lr0=0.01' and 'momentum=0.937' and determining best 'optimizer', 'lr0' and 'momentum' automatically... 
optimizer: AdamW(lr=0.000119, momentum=0.9) with parameter groups 139 weight(decay=0.0), 189 weight(decay=0.0005), 153 bias(decay=0.0)
Image sizes 640 train, 640 val
Using 4 dataloader workers
Logging results to runs\detect\train16
Starting training for 20 epochs...

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
       1/20      3.34G      3.513      5.717      4.266         93        640: 100%|██████████| 16/16 [00:20<00:00,  1.29s/it]
                 Class     Images  Instances      Box(P          R      mAP50      mAP75  mAP50-95): 100%|██████████| 8/8 [00:03<00:00,  2.11it/s]
                   all        128        929          0          0          0          0          0

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
       2/20      3.48G      3.617       5.71      4.264        132        640: 100%|██████████| 16/16 [00:07<00:00,  2.26it/s]
                 Class     Images  Instances      Box(P          R      mAP50      mAP75  mAP50-95): 100%|██████████| 8/8 [00:02<00:00,  2.99it/s]
                   all        128        929          0          0          0          0          0

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
       3/20      3.39G      3.505      5.699       4.24        157        640: 100%|██████████| 16/16 [00:07<00:00,  2.01it/s]
                 Class     Images  Instances      Box(P          R      mAP50      mAP75  mAP50-95): 100%|██████████| 8/8 [00:02<00:00,  2.97it/s]
                   all        128        929          0          0          0          0          0

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
       4/20      3.39G      3.539      5.725      4.275        114        640: 100%|██████████| 16/16 [00:07<00:00,  2.27it/s]
                 Class     Images  Instances      Box(P          R      mAP50      mAP75  mAP50-95): 100%|██████████| 8/8 [00:02<00:00,  3.09it/s]
                   all        128        929          0          0          0          0          0

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
       5/20      3.47G      3.603      5.681      4.273        145        640: 100%|██████████| 16/16 [00:11<00:00,  1.42it/s]
                 Class     Images  Instances      Box(P          R      mAP50      mAP75  mAP50-95): 100%|██████████| 8/8 [00:02<00:00,  3.16it/s]
                   all        128        929          0          0          0          0          0

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
       6/20      3.41G      3.672      5.732      4.253        104        640: 100%|██████████| 16/16 [00:11<00:00,  1.41it/s]
                 Class     Images  Instances      Box(P          R      mAP50      mAP75  mAP50-95): 100%|██████████| 8/8 [00:02<00:00,  3.02it/s]
                   all        128        929          0          0          0          0          0

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
       7/20      3.39G      3.619      5.692      4.263        173        640: 100%|██████████| 16/16 [00:07<00:00,  2.27it/s]
                 Class     Images  Instances      Box(P          R      mAP50      mAP75  mAP50-95): 100%|██████████| 8/8 [00:02<00:00,  3.62it/s]
                   all        128        929          0          0          0          0          0

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
       8/20      3.39G      3.697      5.732      4.238         69        640: 100%|██████████| 16/16 [00:06<00:00,  2.47it/s]
                 Class     Images  Instances      Box(P          R      mAP50      mAP75  mAP50-95): 100%|██████████| 8/8 [00:02<00:00,  3.45it/s]
                   all        128        929          0          0          0          0          0

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
       9/20      3.39G      3.472      5.697      4.254         71        640: 100%|██████████| 16/16 [00:10<00:00,  1.59it/s]
                 Class     Images  Instances      Box(P          R      mAP50      mAP75  mAP50-95): 100%|██████████| 8/8 [00:02<00:00,  3.48it/s]
                   all        128        929          0          0          0          0          0

**Comment by JackyhuangUp** [comment]
I have the same problem as you, but I am using the brain tumor dataset. I guess it is a parameter setting issue, but it has not been resolved yet. I look forward to someone providing a solution

**Comment by Mengqi-Lei** [comment]
> I followed the official tutorial. 1. Train coco128. I started training map=0 from 0. 2. If I load the pre-trained weights of yolov13, the map would be 0.7, but I don't know if the network is trained. 3. Then I used my own dataset to load the yolov13 pre-trained weights, with a map of 0.000 or so, but it couldn't be improved. Why is this?

Thank you for your interest! From the terminal output you shared, we can see that the dataset loads correctly and the training process starts without issues, so your implementation itself is fine. In fact, getting all-zero metrics when training *from scratch* on COCO128 is expected: COCO128 is a validation–demo subset with only 128 images, and a model cannot realistically learn 80 classes’ complex visual features from such a small sample. The same result would occur with any YOLO version. We therefore recommend switching to a dataset with far more images for both training and evaluation.
